xbmc/utils/AMLUtils.cpp

```diff
@@ -41,295 +41,200 @@
-    if(SysfsUtils::GetString("/sys/class/amstream/vcodec_profile", valstr) != 0)
-      has_hevc = 0;
-    else
-      has_hevc = (valstr.find("hevc:") != std::string::npos) ? 1: 0;
+    CLog::Log(LOGERROR, "AML: no rw on /dev/video10");
+    return false;
   }
-  return (has_hevc == 1);
-}
 
-bool aml_support_hevc_4k2k()
-{
-  static int has_hevc_4k2k = -1;
+  if (!SysfsUtils::HasRW("/sys/module/amlvideodri/parameters/freerun_mode"))
+  {
+    CLog::Log(LOGERROR, "AML: no rw on /sys/module/amlvideodri/parameters/freerun_mode");
+    return false;
+  }
 
-  if (has_hevc_4k2k == -1)
+  if (!SysfsUtils::HasRW("/sys/class/video/freerun_mode"))
   {
-    CRegExp regexp;
-    regexp.RegComp("hevc:.*4k");
-    std::string valstr;
-    if (SysfsUtils::GetString("/sys/class/amstream/vcodec_profile", valstr) != 0)
-      has_hevc_4k2k = 0;
-    else
-      has_hevc_4k2k = (regexp.RegFind(valstr) >= 0) ? 1 : 0;
+    CLog::Log(LOGERROR, "AML: no rw on /sys/class/video/freerun_mode");
+    return false;
   }
-  return (has_hevc_4k2k == 1);
-}
 
-bool aml_support_hevc_10bit()
-{
-  static int has_hevc_10bit = -1;
+  if (!SysfsUtils::HasRW("/sys/class/audiodsp/digital_raw"))
+  {
+    CLog::Log(LOGERROR, "AML: no rw on /sys/class/audiodsp/digital_raw");
+    return false;
+  }
 
-  if (has_hevc_10bit == -1)
+  if (!SysfsUtils::HasRW("/sys/class/amhdmitx/amhdmitx0/config"))
   {
-    CRegExp regexp;
-    regexp.RegComp("hevc:.*10bit");
-    std::string valstr;
-    if (SysfsUtils::GetString("/sys/class/amstream/vcodec_profile", valstr) != 0)
-      has_hevc_10bit = 0;
-    else
-      has_hevc_10bit = (regexp.RegFind(valstr) >= 0) ? 1 : 0;
+    CLog::Log(LOGERROR, "AML: no rw on /sys/class/amhdmitx/amhdmitx0/config");
+    return false;
   }
-  return (has_hevc_10bit == 1);
-}
 
-AML_SUPPORT_H264_4K2K aml_support_h264_4k2k()
-{
-  static AML_SUPPORT_H264_4K2K has_h264_4k2k = AML_SUPPORT_H264_4K2K_UNINIT;
+  if (!SysfsUtils::HasRW("/sys/class/vfm/map"))
+  {
+    CLog::Log(LOGERROR, "AML: no rw on /sys/class/vfm/map");
+    return false;
+  }
 
-  if (has_h264_4k2k == AML_SUPPORT_H264_4K2K_UNINIT)
+  if (!SysfsUtils::HasRW("/sys/class/tsync/enable"))
   {
-    std::string valstr;
-    if (SysfsUtils::GetString("/sys/class/amstream/vcodec_profile", valstr) != 0)
-      has_h264_4k2k = AML_NO_H264_4K2K;
-    else if (valstr.find("h264:4k") != std::string::npos)
-      has_h264_4k2k = AML_HAS_H264_4K2K_SAME_PROFILE;
-    else if (valstr.find("h264_4k2k:") != std::string::npos)
-      has_h264_4k2k = AML_HAS_H264_4K2K;
-    else
-      has_h264_4k2k = AML_NO_H264_4K2K;
+    CLog::Log(LOGERROR, "AML: no rw on /sys/class/tsync/enable");
+    return false;
   }
-  return has_h264_4k2k;
-}
 
-bool aml_support_vp9()
-{
-  static int has_vp9 = -1;
+  if (!SysfsUtils::HasRW("/sys/devices/system/cpu/cpu0/cpufreq/scaling_min_freq"))
+  {
+    CLog::Log(LOGERROR, "AML: no rw on /sys/devices/system/cpu/cpu0/cpufreq/scaling_min_freq");
+    return false;
```

Comment:
nope. this changes behaviour. 

having cpufreq enabled is not a requirment for hardware decoding.